buildSrc/src/main/kotlin/elastic-otel.agent-packaging-conventions.gradle.kts

```diff
@@ -81,15 +81,15 @@ tasks {
     relocatePackages(this)
 
     // exclude known bootstrap dependencies - they can't appear in the inst/ directory
-    dependencies {
-      exclude("io.opentelemetry:opentelemetry-api")
-      exclude("io.opentelemetry:opentelemetry-api-incubator")
-      exclude("io.opentelemetry:opentelemetry-context")
-      exclude("io.opentelemetry:opentelemetry-semconv")
-      exclude("io.opentelemetry:opentelemetry-semconv-incubating")
-      // metrics advice API
-      exclude("io.opentelemetry:opentelemetry-extension-incubator")
-    }
+
+    exclude("io/opentelemetry/api/")
+    // SDK should not be included, but we have one extra addition to it
+    exclude { it.path.startsWith("io/opentelemetry/sdk") and !it.name.contains("FieldBackedSpanValueStorageProvider.class") }
+    exclude("io/opentelemetry/semconv/")
+    exclude("io/opentelemetry/context/")
+    exclude("io/opentelemetry/internal/")
+    // metrics advice API
+    exclude("io/opentelemetry/extension/incubator/")
```

Comment:
[for reviewer] For a reason I ignore the exclusion did not work as expected and I had to use file paths to select which classes should be included or not.

This allows to keep `implementation` as dependency scope for dependencies (and not `compileOnly`) and still ensure that we don't accidentally ship API/SDK classes in the bootstrap classloader.